releasing.md

```diff
@@ -0,0 +1,302 @@
+
+Submit your PGP key to https://pgp.mit.edu/ . This is required by Sonatype Nexus!
+
+Find your key and get the armored key:
+```
+$  gpg --list-keys
+
+   /Users/ali/.gnupg/pubring.gpg
+   -----------------------------
+   pub   4096R/10CAEE00 2017-09-25
+   uid       [ultimate] Ali Ok (Ali's GPG key) <aliok@apache.org>
+   sub   4096R/3E608DB2 2017-09-25
+
+$  gpg --armor --export 10CAEE00
+
+    -----BEGIN PGP PUBLIC KEY BLOCK-----
+    Version: GnuPG v2
+
+    mQINBFnI6gsBEAC43ZrS75PAbQCQpvOm70oTMwOPbKaNWqCQLxDR/ck7whBpO9wo
+    ...
+    2rS1FJUwRsqi
+    =CoyC
+    -----END PGP PUBLIC KEY BLOCK-----
+```
+
+Copy the block (including the `--------BEGIN...` and `-----END...` lines) and paste it to https://pgp.mit.edu/ and submit.
+
+#### Requirement - Sonatype Nexus account
+
+1. Create Sonatype account here: https://issues.sonatype.org/secure/Signup!default.jspa
+2. Check if login works to Nexus: https://oss.sonatype.org/
+3. Ask for publishing permissions for the `groupId` (in pom.xml) of the project you want to publish. Ask @passos or @matzew to add a comment to one of these tickets:
+   * `groupId org.aerogear`: https://issues.sonatype.org/browse/OSSRH-34060
+   * `groupId org.jboss.aerogear`: https://issues.sonatype.org/browse/OSSRH-20598
+
+## Setup
+
+1. GPG
+2. Maven Security
+3. Sonatype Nexus
+
+#### Setup - GPG
+
+Make sure GPG works as described in the requirements section.
+Add your gpg secret key id to your global git config : ```git config --global user.signingkey YOURKEYID```
+
+#### Setup - Maven Security
+
+Set up the password encryption for Maven, you can follow these instructions : http://maven.apache.org/guides/mini/guide-encryption.html
+
+Make sure you created a master password.
+
+### Setup - Sonatype Nexus
+
+For the reference, in order to publish to Sonatype OSS repo, one must:
+
+* Make sure your pom.xml is conforming the requirements of Sonatype: http://central.sonatype.org/pages/apache-maven.html
+* Have this section in the pom.xml file:
+
+```
+    <distributionManagement>
+        <snapshotRepository>
+            <id>ossrh</id>
+            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+        </snapshotRepository>
+        <repository>
+            <id>ossrh</id>
+            <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
+        </repository>
+    </distributionManagement>
+```
+Things above are already done, so you don't need to worry.
+
+Encrypt your Sonatype password:
+```
+mvn --encrypt-password <the password>
+> {COQLCE6DU6GtcS5P=}
+```
+
+Use that password in your `~/.m2/settings.xml`:
+
+```
+<settings>
+  <servers>
+    ...
+    <server>
+      <id>ossrh</id>
+      <username>YOUR_USERNAME</username>
+      <password>{COQLCE6DU6GtcS5P=}</password>
+    </server>
+    ...
+  </servers>
+</settings>
+
+```
+
+## Testing by releasing snapshots
+
+You can test if things work correctly by releasing a snapshot.
```

Comment:
Not sure of the use of the wore 'things' here. Maybe use, `The release process can be tested by releasing a snapshot`

Reply:
makes sense